pkgs/applications/office/jabref/default.nix

```
@@ -118,14 +134,17 @@ stdenv.mkDerivation rec {
     tar xf build/distributions/JabRef-${version}.tar -C $out --strip-components=1
     unzip $out/lib/javafx-web-*-linux${lib.optionalString stdenv.isAarch64 "-aarch64"}.jar libjfxwebkit.so -d $out/lib/
 
+    rm $out/lib/javafx-*-win.jar
+
     DEFAULT_JVM_OPTS=$(sed -n -E "s/^DEFAULT_JVM_OPTS='(.*)'$/\1/p" $out/bin/JabRef | sed -e "s|\$APP_HOME|$out|g" -e 's/"//g')
     rm $out/bin/*
 
     makeWrapper ${jdk}/bin/java $out/bin/JabRef \
       "''${gappsWrapperArgs[@]}" \
       --suffix PATH : ${lib.makeBinPath [ xdg-utils ]} \
       --add-flags "-Djava.library.path=$out/lib/ --patch-module org.jabref=$out/share/java/jabref/resources/main" \
-      --add-flags "$DEFAULT_JVM_OPTS"
+      --add-flags "$DEFAULT_JVM_OPTS" \
+      --prefix LD_LIBRARY_PATH : ${rpath}
```

```suggestion
      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libXxf86vm libXtst ]}
```
but more importantly: did you try to use patchelf?

---

Those .so files are in jars. I guess if I want to patchelf I need to unpack the jar then repack it. I'll have a try.